app/components/FilterableComponents/FilterableTableLayout.tsx

```diff
@@ -1,29 +1,81 @@
-import React from 'react';
+import React, {useMemo} from 'react';
 import {Col, Table, Alert} from 'react-bootstrap';
 import SortableTableHeader from './SortableTableHeader';
 import FilterableTableHeaders from './FilterableTableHeaders';
-import {ISearchProps} from 'components/Search/SearchProps';
+import {ISearchProps, ISearchExtraFilter} from 'components/Search/SearchProps';
+import {useRouter} from 'next/router';
+import safeJsonParse from 'lib/safeJsonParse';
+import {FilterArgs} from 'components/Search/ISearchOption';
 
 interface Props extends ISearchProps {
   body: JSX.Element;
   isLoading?: boolean;
-  extraControls?: JSX.Element;
+  extraFilters?: ISearchExtraFilter[];
 }
-export const FilterableTableLayoutComponent: React.FunctionComponent<Props> = (
-  props
-) => {
-  const {searchOptions, body, isLoading} = props;
+export const FilterableTableLayoutComponent: React.FunctionComponent<Props> = ({
+  searchOptions,
+  body,
+  isLoading,
+  extraFilters = []
+}) => {
+  const router = useRouter();
+  const relayVars = useMemo(
+    () => safeJsonParse(router.query.relayVars as string),
+    [router]
```

Comment:
I'm not really familiar with how useMemo detects changes, isn't the router object going to be the same every time, and this won't trigger when the relayVars change?